core/templates/dev/head/pages/exploration-editor-page/improvements-tab/suggestion-improvement-task/suggestion-improvement-task.directive.ts

```diff
@@ -30,38 +30,27 @@ angular.module('oppia').directive('suggestionImprovementTask', [
         'suggestion-improvement-task/' +
         'suggestion-improvement-task.directive.html'),
       controller: [
-        '$scope', 'DateTimeFormatService', 'ThreadStatusDisplayService',
-        function($scope, DateTimeFormatService, ThreadStatusDisplayService) {
+        '$scope', 'DateTimeFormatService',
+        function($scope, DateTimeFormatService) {
           $scope.getLatestMessage = function() {
-            var numMessages = $scope.getData().messages.length;
-            if (numMessages > 1) {
-              var latestMessage = $scope.getData().messages[numMessages - 1];
-              return {
-                text: latestMessage.text,
-                author: latestMessage.author_username,
-                updatedOn: latestMessage.created_on,
-                updatedStatus: latestMessage.updated_status,
-              };
-            } else {
-              return {
-                text: $scope.getData().subject,
-                author: $scope.getData().originalAuthorName,
-                updatedOn: $scope.getData().lastUpdated,
-                updatedStatus: null,
-              };
-            }
-          };
-
-          $scope.getHumanReadableUpdatedStatus = function() {
-            var updatedStatus = $scope.getLatestMessage().updatedStatus;
-            return updatedStatus === null ? null :
-              ThreadStatusDisplayService.getHumanReadableStatus(updatedStatus);
+            var thread = $scope.getData();
+            return {
+              text: thread.lastNonemptyMessageText || thread.subject,
+              author: (
+                thread.lastNonemptyMessageAuthor || thread.originalAuthorName),
+              updatedOn: thread.lastUpdated,
+            };
           };
 
           $scope.getLocaleAbbreviatedDatetimeString = function() {
             return DateTimeFormatService.getLocaleAbbreviatedDatetimeString(
               $scope.getLatestMessage().updatedOn);
           };
+
+          $scope.wasUpdatedToday = function() {
```

Comment:
Hm, I remember seeing this added in a previous PR. Are we duplicating code -- if so, should we instead reuse the previous implementation?